tests/measure/test_metric.py

```diff
@@ -27,3 +29,34 @@ def test_metric(metric_fn: Callable[[Space, Space], torch.Tensor], same_shape_sp
     obj_result = MetricFn(key="test", fn=metric_fn)(space1, space2)["test"]
 
     assert torch.allclose(fn_result, obj_result)
+
+
+@pytest.mark.parametrize(
+    "mode",
+    [
+        (CKAMode.LINEAR),
+        (CKAMode.RBF),
+    ],
+)
+def test_cka(mode: CKAMode, same_shape_spaces, different_dim_spaces):
+
+    for spaces in [same_shape_spaces, different_dim_spaces]:
+        space1, space2 = spaces
+        cka = CKA(mode=mode)
+        cka_result = cka(space1, space2)
+
+        # cka must stay in 0, 1 range
+        assert 0.0 - TOL <= cka_result <= 1.0 + TOL
+
+        # cka is symmetric
+        symm_cka_result = cka(space2, space1)
+        assert symm_cka_result == pytest.approx(cka_result, abs=TOL)
+
+        # check that GPU works correctly
+        cka_gpu = CKA(mode=mode, device=torch.device("cuda"))
```

Comment:
Just as a sanity check I would also check the `device=None` and then `cka_gpu.to('cuda')` works